### PR TITLE
VC Activity command crash fix #300

### LINKED ISSUE #300: VC Activity command crashes
**Describe the bug**
The vc activity command crashes when the max-age exceeds about 1 million.

**Screenshots**
image

--- 

This could be fixed by, for example, implementing a max value for the max-age.

### PR DESCRIPTION
**Pull request by Budbomber**
Set a max duration of a week in seconds, (Not sure what the best way to do it was, so went with a 60*60*24) 7 days in seconds.

Added required range to MAX_AGE_OPTION (0, MAX_AGE_DURATION_ONE_WEEK).

### CODE REVIEW

**Review by Tais993 — CHANGES_REQUESTED** [review]

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java [hunk 1]
```diff
@@ -45,6 +45,9 @@
     private static final String MAX_USES_OPTION = "max-uses";
     private static final String MAX_AGE_OPTION = "max-age";
 
+    //The amount of seconds in one week.
+    private static final int MAX_AGE_DURATION_ONE_WEEK = 60 * 60 * 24;
```
[the remark this answers is not in the document]
Comment (Budbomber) [reply application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:49]: Ahh, this is where I got it wrong, I Was originally trying to use TimeUnit, and doing some funny things with it.

**Comment by Zabuzard** [comment]
Apart from what has been mentioned, looks fine. That said, what about the rest of the code? If I remember correctly, it had some check (`handleIntegerOption` or sth like that) which might now be (partially?) obsolete?

Please double check and if necessary clean up, thanks.

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java [hunk 2]
```diff
@@ -86,8 +89,8 @@
             new OptionData(OptionType.STRING, MAX_USES_OPTION,
                     "The amount of times the invite can be used, default is infinity", false),
             new OptionData(OptionType.INTEGER, MAX_AGE_OPTION,
-                    "Max age in seconds. Set this to 0 to never expire, default is 1 day", false));
-
+                    "Max age in seconds. Set this to 0 to never expire, default is 1 day", false)
+                        .setRequiredRange(0, MAX_AGE_DURATION_ONE_WEEK)); // Max duration 0 - 7 days in seconds.
```
[the remark this answers is not in the document]
Comment (Budbomber) [reply application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:92]: Yep, max sense thank you Tais

**Comment by Budbomber** [comment]
> 

This would also mean changing max users to the same effect as age, and the removing that handler totally

**Review by Budbomber — COMMENTED** [review]
More readable following changes, accepted suggested.

**Comment by Budbomber** [comment]
> Set a max duration of a week in seconds, (Not sure what the best way to do it was, so went with a 60_60_24) 7 days in seconds.
> 
> Added required range to MAX_AGE_OPTION (0, MAX_AGE_DURATION_ONE_WEEK).

This would also mean changing max users to the same effect as age, and the removing that handler totally

**Comment by Zabuzard** [comment]
> > Set a max duration of a week in seconds, (Not sure what the best way to do it was, so went with a 60_60_24) 7 days in seconds.
> > Added required range to MAX_AGE_OPTION (0, MAX_AGE_DURATION_ONE_WEEK).
> 
> This would also mean changing max users to the same effect as age, and the removing that handler totally

Thats fair, but what about this part of the code:
https://i.imgur.com/aCiAplh.png
Is it really still necessary like that, or can it be simplified now?

**Comment by Budbomber** [comment]
> > > Set a max duration of a week in seconds, (Not sure what the best way to do it was, so went with a 60_60_24) 7 days in seconds.
> > > Added required range to MAX_AGE_OPTION (0, MAX_AGE_DURATION_ONE_WEEK).
> > 
> > 
> > This would also mean changing max users to the same effect as age, and the removing that handler totally
> 
> Thats fair, but what about this part of the code:  Is it really still necessary like that, or can it be simplified now?

Pretty sure we can just remove that, plus the calls to the error handlers

**Comment by Tais993** [comment]
I see that there also is a max to the max-uses option, see the Discord docs [here](https://discord.com/developers/docs/resources/channel#create-channel-invite)

Has to be between 0 and 100, fix that too in this PR?

That way the handleIntegerTypeOption's validation regarding it being a valid Integer can be removed

**Comment by Budbomber** [comment]
> > 
> 
> No worries, I'll work on that now and clean up it up

Seems to be working now, changes made, any input appreciated

**Review by Budbomber — COMMENTED** [review]
Perfect

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java [hunk 3]
```diff
@@ -45,6 +45,10 @@
     private static final String MAX_USES_OPTION = "max-uses";
     private static final String MAX_AGE_OPTION = "max-age";
 
+    // sets the Max_age duration of the voice channel in seconds - converts to days.
```
Comment (illuminator3) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:48]:
```suggestion
    /**
     * Sets the MAX_AGE duration of the voice channel in seconds - converts to days.
     */
```

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java [hunk 4]
```diff
@@ -83,11 +87,12 @@
...
-            new OptionData(OptionType.STRING, MAX_USES_OPTION,
-                    "The amount of times the invite can be used, default is infinity", false),
+            new OptionData(OptionType.INTEGER, MAX_USES_OPTION,
+                    "The amount of times the invite can be used, default is infinity", false)
+                    .setRequiredRange(0, MAX_VALE_MAX_USES),
             new OptionData(OptionType.INTEGER, MAX_AGE_OPTION,
-                    "Max age in seconds. Set this to 0 to never expire, default is 1 day", false));
-
+                    "Max age in seconds. Set this to 0 to never expire, default is 1 day", false)
+                        .setRequiredRange(1, MAX_VALUE_MAX_AGE));
```
Comment (illuminator3) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:95]:
```suggestion
                        .setRequiredRange(0, MAX_VALUE_MAX_AGE));
```

**Review by illuminator3 — CHANGES_REQUESTED** [review]

**Review by illuminator3 — DISMISSED** [review]
LGTM

**Comment by Tais993** [comment]
Please don't merge this PR till I reviewed ^^

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java [hunk 5]
```diff
@@ -45,6 +45,12 @@
     private static final String MAX_USES_OPTION = "max-uses";
     private static final String MAX_AGE_OPTION = "max-age";
 
+    /**
+     * Sets the MAX_AGE duration of the voice channel in seconds - converts to days.
+     */
```
Comment (Zabuzard) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:50]: The comment is kinda obsolete and doesnt add anything to the code, I would remove it.

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java [hunk 6]
```diff
@@ -222,52 +204,4 @@ private static void handleErrors(@NotNull SlashCommandEvent event,
...
-            @Nullable OptionMapping optionMapping) {
-
-        int optionValue;
-
-        if (optionMapping == null) {
-            return null;
-        }
-
-        try {
-            optionValue = Math.toIntExact(optionMapping.getAsLong());
```
Comment (Tais993) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:252]: Keep this too

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java [hunk 7]
```diff
@@ -222,52 +204,4 @@ private static void handleErrors(@NotNull SlashCommandEvent event,
...
-     */
-    @Contract("_, null -> null")
-    private static @Nullable Integer handleIntegerTypeOption(@NotNull SlashCommandEvent event,
-            @Nullable OptionMapping optionMapping) {
-
-        int optionValue;
-
-        if (optionMapping == null) {
-            return null;
-        }
```
Comment (Tais993) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:249]: This still is an important part, the value is optional, not required, so if it's not provided it should be null

**Review by Tais993 — CHANGES_REQUESTED** [review]

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java (hunk 6 above)
Comment (Budbomber) [reply application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:252]: I've done it slightly differently, hopefully, it has the same effect? seems to work fine now.

Comment (Tais993) [reply application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:252]: You've done it exactly like how I wanted it to be done, you keep that line and remove the rest

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java [hunk 8]
```diff
@@ -223,51 +228,19 @@ private static void handleErrors(@NotNull SlashCommandEvent event,
...
-     * @param event the {@link SlashCommandEvent}
-     * @param optionMapping the {@link OptionMapping}
-     * @return nullable {@link Integer}
-     * @throws java.lang.IllegalArgumentException if the option's value is - outside of
-     *         {@link Integer#MAX_VALUE} - negative
-     */
-    @Contract("_, null -> null")
-    private static @Nullable Integer handleIntegerTypeOption(@NotNull SlashCommandEvent event,
+    @Contract("null -> null")
+    private static @Nullable Integer handleIntegerTypeOption(
```
Comment (Tais993) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:232]:
```suggestion
    /**
     * This grabs the OptionMapping, after this it returns null of the OptionMapping is null, else it'll return the number option as an Integer
     *
     * <p>
     * <p/>
     *
     * @param optionMapping the {@link OptionMapping}
     * @return nullable {@link Integer}
     **/
    @Contract("null -> null")
    private static @Nullable Integer handleIntegerTypeOption(
```

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java [hunk 9]
```diff
@@ -148,43 +155,40 @@ public void onSlashCommand(@NotNull SlashCommandEvent event) {
...
+            applicationId = Objects.requireNonNull(VC_APPLICATION_TO_ID.get(applicationName));
         } else {
-            applicationId = idOption.getAsString();
 
-            applicationName =
-                    getKeyByValue(VC_APPLICATION_TO_ID, applicationId).orElse("an activity");
+            applicationId = (Objects.requireNonNull(idOption.getAsString()));
+
+            applicationName = (Objects.requireNonNull(
+                    getKeyByValue(VC_APPLICATION_TO_ID, applicationId).orElse("an activity")));
```
Comment (Tais993) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:183]: Why did you change this?

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java [hunk 10]
```diff
@@ -148,43 +155,40 @@ public void onSlashCommand(@NotNull SlashCommandEvent event) {
...
         }
 
-
-        String applicationId;
-        String applicationName;
-
         if (applicationOption != null) {
             applicationName = applicationOption.getAsString();
-            applicationId = VC_APPLICATION_TO_ID.get(applicationName);
+            applicationId = Objects.requireNonNull(VC_APPLICATION_TO_ID.get(applicationName));
```
Comment (Tais993) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:177]: And this?

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java [hunk 11]
```diff
@@ -148,43 +155,40 @@ public void onSlashCommand(@NotNull SlashCommandEvent event) {
...
-
-        // the user already received the error in the handleIntegerTypeOption method
-        // it still throws to tell us to return this method and stop the proceeding code
         try {
-            maxAge = handleIntegerTypeOption(event, maxAgeOption);
-        } catch (IllegalArgumentException ignore) {
+            maxAge = handleIntegerTypeOption(maxAgeOption);
+        } catch (IllegalArgumentException ignored) {
             return;
         }
```
Comment (Tais993) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:173]: The handleIntegerTypeOption doesn't throw, so the try-catch can be removed

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java [hunk 12]
```diff
@@ -148,43 +155,40 @@ public void onSlashCommand(@NotNull SlashCommandEvent event) {
...
 
-        // the user already received the error in the handleIntegerTypeOption method
-        // it still throws to tell us to return this method and stop the proceeding code
         try {
-            maxUses = handleIntegerTypeOption(event, maxUsesOption);
-        } catch (IllegalArgumentException ignore) {
+            maxUses = handleIntegerTypeOption(maxUsesOption);
+        } catch (IllegalArgumentException ignored) {
             return;
         }
```
Comment (Tais993) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:167]: The handleIntegerTypeOption doesn't throw, so the try-catch can be removed

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java [hunk 13]
```diff
@@ -100,17 +105,19 @@ public VcActivityCommand() {
...
+                    .addOptions(inviteOptions));
+
 
 
-        SubcommandData idSubCommand =
-                new SubcommandData("id", "specify the ID for the application manually")
-                    .addOption(OptionType.STRING, ID_OPTION, "the ID of the application", true)
-                    .addOptions(inviteOptions);
+        SubcommandData idSubCommand = Objects
+            .requireNonNull(new SubcommandData("id", "specify the ID for the application manually")
```
Comment (Tais993) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:117]: Why did you change this?

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java [hunk 14]
```diff
@@ -100,17 +105,19 @@ public VcActivityCommand() {
                 SlashCommandVisibility.GUILD);
 
 
-        SubcommandData applicationSubCommand =
+        SubcommandData applicationSubCommand = Objects.requireNonNull(
```
Comment (Tais993) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:108]: Why did you change this?

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java [hunk 15]
```diff
@@ -45,6 +46,9 @@
     private static final String MAX_USES_OPTION = "max-uses";
     private static final String MAX_AGE_OPTION = "max-age";
 
+    private static final long MAX_VALUE_MAX_AGE = TimeUnit.SECONDS.toDays(604800);
```
Comment (Tais993) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:49]:
```suggestion
    private static final long MAX_VALUE_MAX_AGE = TimeUnit.DAYS.toSeconds(7);
```
604800 is still unreadable, I'm too lazy to grab my calculator to find out how many days that is.
We wanted to use the TimeUnit exactly so the 604800 could get removed.

**Review by Tais993 — CHANGES_REQUESTED** [review]
You added a lot of random Object#requireNonNull's, even when the return value is `@NonNull`

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java [hunk 16]
```diff
@@ -148,43 +155,33 @@ public void onSlashCommand(@NotNull SlashCommandEvent event) {
...
+        maxAge = handleIntegerTypeOption(maxAgeOption);
 
-        String applicationId;
-        String applicationName;
 
         if (applicationOption != null) {
             applicationName = applicationOption.getAsString();
             applicationId = VC_APPLICATION_TO_ID.get(applicationName);
         } else {
+
```
Comment (Tais993) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:172]:
```suggestion
```

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java [hunk 17]
```diff
@@ -107,12 +112,14 @@ public VcActivityCommand() {
...
 
 
+
         SubcommandData idSubCommand =
                 new SubcommandData("id", "specify the ID for the application manually")
                     .addOption(OptionType.STRING, ID_OPTION, "the ID of the application", true)
                     .addOptions(inviteOptions);
 
 
+
```
Comment (Tais993) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:122]:
```suggestion
```

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java [hunk 18]
```diff
@@ -107,12 +112,14 @@ public VcActivityCommand() {
...
```
Comment (Tais993) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:115]:
```suggestion
```

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java [hunk 19]
```diff
@@ -148,43 +155,33 @@ public void onSlashCommand(@NotNull SlashCommandEvent event) {
...
             applicationId = VC_APPLICATION_TO_ID.get(applicationName);
         } else {
+
             applicationId = idOption.getAsString();
 
             applicationName =
                     getKeyByValue(VC_APPLICATION_TO_ID, applicationId).orElse("an activity");
         }
 
+
```
Comment (Tais993) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:179]:
```suggestion
```

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java [hunk 20]
```diff
@@ -148,43 +155,33 @@ public void onSlashCommand(@NotNull SlashCommandEvent event) {
...
         handleSubcommand(event, voiceChannel, applicationId, maxUses, maxAge, applicationName);
     }
 
+
+
```
Comment (Tais993) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:184]:
```suggestion
```

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java [hunk 21]
```diff
@@ -200,6 +197,7 @@ private static void handleSubcommand(@NotNull SlashCommandEvent event,
             @NotNull VoiceChannel voiceChannel, @NotNull String applicationId,
             @Nullable Integer maxUses, @Nullable Integer maxAge, @NotNull String applicationName) {
 
+
```
Comment (Tais993) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:200]:
```suggestion
```

**Review by Tais993 — COMMENTED** [review]
I personally don't like all the new lines you added, it's too many of them imo

Besides that, it looks good

**Comment by Budbomber** [comment]
> 

Yeah sorry bud must of been the merges etc I failed haha next PR I will do much neater haha

**Review by Tais993 — APPROVED** [review]

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java [hunk 22]
```diff
@@ -83,11 +89,12 @@
             "879863976006127627", LETTERTILE_NAME, "879863686565621790");
 
     private static final List<OptionData> inviteOptions = List.of(
-            new OptionData(OptionType.STRING, MAX_USES_OPTION,
-                    "The amount of times the invite can be used, default is infinity", false),
+            new OptionData(OptionType.INTEGER, MAX_USES_OPTION,
+                    "The amount of times the invite can be used, default is infinity", false)
+                    .setRequiredRange(0, MAX_VALE_MAX_USES),
```
[the remark this answers is not in the document]
Comment (Budbomber) [reply application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java]: 0 is default value for both just an error on my part

**Review by illuminator3 — APPROVED** [review]
Approving this as Tijs has also approved it.

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java [hunk 23]
```diff
@@ -148,29 +153,15 @@ public void onSlashCommand(@NotNull SlashCommandEvent event) {
...
-        // the user already received the error in the handleIntegerTypeOption method
-        // it still throws to tell us to return this method and stop the proceeding code
-        try {
-            maxAge = handleIntegerTypeOption(event, maxAgeOption);
-        } catch (IllegalArgumentException ignore) {
-            return;
-        }
 
+        maxUses = handleIntegerTypeOption(maxUsesOption);
+        maxAge = handleIntegerTypeOption(maxAgeOption);
```
Comment (Zabuzard) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:163]: Assign the variables directly, dont declare and then a line later assign.

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java [hunk 24]
```diff
@@ -148,29 +153,15 @@ public void onSlashCommand(@NotNull SlashCommandEvent event) {
         OptionMapping maxUsesOption = event.getOption(MAX_USES_OPTION);
         OptionMapping maxAgeOption = event.getOption(MAX_AGE_OPTION);
 
+        String applicationId;
+        String applicationName;
```
Comment (Zabuzard) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:157]: Dont declare variables before they are needed. Put this as late as possible, i.e.  at line 165.

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java [hunk 25]
```diff
@@ -223,51 +214,29 @@ private static void handleErrors(@NotNull SlashCommandEvent event,
         logger.warn("Something went wrong in the VcActivityCommand", throwable);
     }
 
-
     /**
-     * This grabs the OptionMapping, after this it <br />
-     * - validates whenever it's within an {@link Integer Integer's} range <br />
-     * - validates whenever it's positive <br />
+     * This grabs the OptionMapping, after this it returns null of the OptionMapping is null, else
+     * it'll return the number option as an Integer
```
Comment (Zabuzard) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:219]: `if` not `of`. That said, this javadoc rather explains the implementation and not the intention or idea behind the method.

Maybe reword to `Interprets the given option as integer. Throws if the option is not an integer.` And then `@return the option interpreted as `int`, `null` if the option was `null` already.

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java [hunk 26]
```diff
@@ -223,51 +214,29 @@ private static void handleErrors(@NotNull SlashCommandEvent event,
...
      * @param optionMapping the {@link OptionMapping}
      * @return nullable {@link Integer}
-     * @throws java.lang.IllegalArgumentException if the option's value is - outside of
-     *         {@link Integer#MAX_VALUE} - negative
-     */
-    @Contract("_, null -> null")
-    private static @Nullable Integer handleIntegerTypeOption(@NotNull SlashCommandEvent event,
+     **/
+    @Contract("null -> null")
+    private static @Nullable Integer handleIntegerTypeOption(
```
Comment (Zabuzard) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:228]: With all the error handling removed, this method is obsolete. Inline the logic and remove the method.

**Review by Zabuzard — CHANGES_REQUESTED** [review]

**Comment by Zabuzard** [comment]
@Budbomber Please dont resolve conversations yourself. Let them open and wait until the person who requested change approves and closes them, thanks.

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java [hunk 27]
```diff
@@ -83,11 +87,12 @@
             "879863976006127627", LETTERTILE_NAME, "879863686565621790");
 
     private static final List<OptionData> inviteOptions = List.of(
-            new OptionData(OptionType.STRING, MAX_USES_OPTION,
-                    "The amount of times the invite can be used, default is infinity", false),
+            new OptionData(OptionType.INTEGER, MAX_USES_OPTION,
+                    "The amount of times the invite can be used, default is infinity", false)
+                        .setRequiredRange(0, MAX_VALE_MAX_USES),
```
[the remark this answers is not in the document]
Comment (Budbomber) [reply application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:92]: New descriptions, -                     "Max uses for invites, 0 for infinite, 100 is the top limit.", false)

-                     "Max age in seconds. Set this to 0 to never expire, default is 1 day", false)

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java (hunk 23 above)
Comment (Budbomber) [reply application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:163]: Direct assignment

 Integer maxUses = handleIntegerTypeOption(maxUsesOption);
 Integer maxAge = handleIntegerTypeOption(maxAgeOption);

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java (hunk 24 above)
Comment (Budbomber) [reply application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:157]: Moved these now.

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java (hunk 27 above)
[the remark this answers is not in the document]
Comment (Zabuzard) [reply application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:92]: "max uses for invites" and "max age in seconds" is a bad description, it doesnt add much that isnt already part of the option name. instead, describe the purpose in more detail.

example:
> how often the invite link to this activity can be used, 0 for infinite (default)
> how long, in seconds, the invite link to this activity can be used before it expires, 0 for never expire, the default is 1 day

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java (hunk 26 above)
Comment (Zabuzard) [reply application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:223]: Or at least rename it, its not a "handle" method anymore. `requireIntOptionIfPresent` and clean up the code
```java
return option == null ? null : Math.toIntExact(option.getAsLong());
```
and also the javadoc then, its written a bit flopsy.

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java (hunk 27 above)
[the remark this answers is not in the document]
Comment (Budbomber) [reply application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java]: I hate how spotless has left this chunk looking now, 

image

The age is now in days as input, so you can input between 0 - 7

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java (hunk 26 above)
Comment (Budbomber) [reply application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java]: Thank you for the guidance, resolved it.

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java [hunk 28]
```diff
@@ -85,13 +85,13 @@
...
-                        .setRequiredRange(0, MAX_USES_LIMIT),
+    private static final List<OptionData> inviteOptions = List.of(new OptionData(OptionType.INTEGER,
+            MAX_USES_OPTION,
+            "How many times this invite can be used, 0 infinite (default) - 100 being the highest.",
+            false).setRequiredRange(0, MAX_USES_LIMIT),
             new OptionData(OptionType.INTEGER, MAX_AGE_OPTION,
-                    "Max age in seconds. Set this to 0 to never expire, default is 1 day", false)
-                        .setRequiredRange(0, MAX_AGE_LIMIT));
+                    "How long, in days this activity can be used before it expires, 0 (No expiry), Maximum is 7 days. ",
+                    false).setRequiredRange(0, MAX_AGE_LIMIT));
```
Comment (Zabuzard) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:94]: The variable name should include the unit to prevent bugs `MAX_AGE_DAYS_LIMIT`

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java [hunk 29]
```diff
@@ -154,8 +154,8 @@ public void onSlashCommand(@NotNull SlashCommandEvent event) {
 
         String applicationId;
         String applicationName;
-        Integer maxUses = handleIntegerTypeOption(maxUsesOption);
-        Integer maxAge = handleIntegerTypeOption(maxAgeOption);
+        Integer maxUses = requiredIntOptionIfPresent(maxUsesOption);
+        Integer maxAge = requiredIntOptionIfPresent(maxAgeOption);
```
Comment (Zabuzard) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:158]: The variable name should include the unit to prevent bugs `maxAgeDays`

Comment (Zabuzard) [reply application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:158]: Now that you made it days, the code will have a bug when you interpret it without additional logic, since you hand the input 1:1 over to `.setMaxAge(maxAge)`, which expects seconds (no?).

**Review by Zabuzard — CHANGES_REQUESTED** [review]

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java (hunk 28 above)
Comment (Budbomber) [reply application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:94]: Changed variable name.

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java (hunk 29 above)
Comment (Budbomber) [reply application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:158]: What's the best way of doing this? Should I convert the units from days to seconds? as you're right and discord expects seconds.

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java [hunk 30]
```diff
@@ -82,12 +85,13 @@
...
-                    "The amount of times the invite can be used, default is infinity", false),
+    private static final List<OptionData> inviteOptions = List.of(new OptionData(OptionType.INTEGER,
+            MAX_USES_OPTION,
+            "How many times this invite can be used, 0 infinite (default) - 100 being the highest.",
+            false).setRequiredRange(0, MAX_USES_LIMIT),
             new OptionData(OptionType.INTEGER, MAX_AGE_OPTION,
-                    "Max age in seconds. Set this to 0 to never expire, default is 1 day", false));
-
+                    "How long, in days this activity can be used before it expires, 0 (No expiry), Maximum is 7 days. ",
+                    false).setRequiredRange(0, MAX_AGE_DAYS_LIMIT));
```
Comment (Zabuzard) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:94]: You wrote `100` and `7` in the text directly. But what if someone changes the constants (lets say 50 and 6), I bet they would forget to also adjust the text...

It would be better to let the text use the variables instead `"... - " + MAX_USES_LIMIT + " being the highest."` etc. If it starts to become unreadable, use `String.format("... %d ...", foo)` instead.

**Review by Zabuzard — CHANGES_REQUESTED** [review]

**Review by Zabuzard — APPROVED** [review]

**Review by Zabuzard — APPROVED** [review]

**Review by Tais993 — APPROVED** [review]
image
